routes/hydrology.py

```diff
@@ -171,43 +179,148 @@ def run_fetch_hydrology_point_data_mmm(lat, lon):
-                        era_name = dim_encodings["eras"][era_coord]
+                        min_value, mean_value, max_value = (
+                            min(values),
+                            round(np.nanmean(values), 2),
+                            max(values),
+                        )
 
-                        values.append(
-                            float(
-                                point_pkg[model_name][scenario_name][month_name][
-                                    era_name
-                                ][var_name]
+                        # reformat NaN stats to string "nan" to avoid "is not valid JSON" error
+
+                        if np.isnan(min_value):
+                            min_value = "nan"
+                        if np.isnan(mean_value):
+                            mean_value = "nan"
+                        if np.isnan(max_value):
+                            max_value = "nan"
+
+                        point_pkg_mmm[model_name][scenario_name][month_name][var_name][
+                            "min"
+                        ] = min_value
+                        point_pkg_mmm[model_name][scenario_name][month_name][var_name][
+                            "mean"
+                        ] = mean_value
+                        point_pkg_mmm[model_name][scenario_name][month_name][var_name][
+                            "max"
+                        ] = max_value
+
+            # Generates the annual min-mean-max for ArcticEDS
+            if summarize:
+                point_pkg_mmm[model_name][scenario_name]['Annual'] = dict()
+                for era_title in dim_encodings["eds_eras"].keys():
+                    for var_coord in dim_encodings["varnames"].keys():
+                        var_name = dim_encodings["varnames"][var_coord]
+                        values = list()
+
+                        # We have to pull the data in this way to ensure we are getting
+                        # the mean for each variable for each month.
+                        for month_coord in dim_encodings["months"].keys():
+                            month_name = dim_encodings["months"][month_coord]
+                            values.append(
+                                float(
+                                    point_pkg_mmm[model_name][scenario_name][month_name][
+                                        var_name
+                                    ][era_title]['mean']
+                                )
                             )
+
+                        # This will be the min, mean, and max taken from the mean monthly values
+                        # to find the min-mean-max of the annual values for a given time period such
+                        # as 1950-2009 for historical.
+                        min_value, mean_value, max_value = (
+                            min(values),
+                            round(np.nanmean(values), 2),
+                            max(values),
```

Comment:
I noticed that if you put a `print(len(values))` before this chunk of code, it shows that the `values` list has 12 values. I think this should be anywhere from 36-72 values (12 months * 3 decades for projected, or 12 months * 6 decades for historical), right?

Reply:
No, that should have only 12 values as those are the 12 mean values for each month for the particular "era" which is something like: historical, early_century, mid_century, and late century by that point in the script. That is the case due to the code in 184 - 224 that generates the monthly based mmm for each era.

Reply:
That might be better clarified with some additional comments.